tfhe/src/core_crypto/experimental/entities/lwe_shrinking_keyswitch_key.rs

```diff
@@ -0,0 +1,328 @@
+//! Module containing the definition of the [`LweShrinkingKeyswitchKey`].
+
+use crate::core_crypto::commons::parameters::*;
+use crate::core_crypto::commons::traits::*;
+use crate::core_crypto::entities::*;
+use crate::core_crypto::experimental::commons::parameters::{
+    LweSecretKeySharedCoefCount, LweSecretKeyUnsharedCoefCount,
+};
+
+/// An [`LWE shrinking keyswitch key`](`LweShrinkingKeyswitchKey`) is an [`LWE keyswitch
+/// key`](`LweKeyswitchKey`) where the input and output key share a part of their randomness.
+///
+/// See [`the formal definition of an LWE keyswitch key`](`LweKeyswitchKey#formal-definition`).
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub struct LweShrinkingKeyswitchKey<C: Container>
+where
+    C::Element: UnsignedInteger,
+{
+    lwe_ksk: LweKeyswitchKey<C>,
+    input_key_lwe_dimension: LweDimension,
```

Comment:
I think this is stored in `lwe_ksk`

Reply:
no the input lwe dim of the lwe_ksk is the unshared randomness

Reply:
yep that sounds about right